validator/src/main.rs

```diff
@@ -991,6 +991,18 @@ pub fn main() {
                 .help("The shred storage size in bytes. \
                        The suggested value is 50% of your ledger storage size in bytes."),
         )
+        .arg(
+            Arg::with_name("ledger_compression")
+                .hidden(true)
+                .long("ledger-compression")
+                .value_name("COMPRESSION_TYPE")
+                .takes_value(true)
+                .possible_values(&["none", "lz4", "snappy", "zlib"])
+                .default_value("none")
+                .help("The compression alrogithm that is used to compress \
```

Comment:
A couple questions:
1) Can columns have some compressed / some uncompressed data, say, if a node has an existing ledger of uncompressed data, but then starts pushing compressed in?
2) Related to above, can someone toggle compression on/off without issue? I assume data will stay compressed in the SST's, but does Rocks have metadata to know what is compressed and what isn't and serve it properly?
3) I'm not sure this is required here, but in the change where you added `rocksdb_shred_compaction`; you added an "EXPERIMENTAL" note? Is that a standard thing we do with new args?

Reply:
3. We'd normally just mark such arguments as hidden

Reply:
> Can columns have some compressed / some uncompressed data, say, if a node has an existing ledger of uncompressed data, but then starts pushing compressed in?

Yes.  RocksDB does encode the compression type to its SST file so it's backward compatible (meaning existing SST files will be left untouched, but newer SST files will be using the compression).

And since the four compression algorithms I included in this PR are all the supported ones from the [default `rust-rocksdb`](https://github.com/yhchiang-sol/rust-rocksdb/blob/master/Cargo.toml#L20), there should be no additional dependency required.

> Related to above, can someone toggle compression on/off without issue? I assume data will stay compressed in the SST's, but does Rocks have metadata to know what is compressed and what isn't and serve it properly?

Yep, they won't have issues as mentioned above.

> We'd normally just mark such arguments as hidden

That sounds like a good plan.  Will do!
And since it's introduced as a hidden argument and we are not changing the default yet, maybe we don't need to worry too much about all areas of possible while enabling more metrics to allow us to track its performance?